app/controllers/api/v0/posts_controller.rb

```diff
@@ -0,0 +1,33 @@
+module Api
+  module V0
+    class PostsController < Api::V0::BaseController
+      include PostsHelper
+
+      before_action only: :show do
+        require_access_token Api::OpenidConnect::Scope.find_by(name: "read")
+      end
+
+      before_action only: %i(create destroy) do
+        require_access_token Api::OpenidConnect::Scope.find_by(name: "read"),
+                             Api::OpenidConnect::Scope.find_by(name: "write")
+      end
+
+      def show
+        posts_services = PostService.new(id: params[:id], user: current_user)
+        posts_services.mark_user_notifications
+        render json: posts_services.present_json
+      end
+
+      def create
+        @status_message = StatusMessageCreationService.new(params, current_user).status_message
+        render json: PostPresenter.new(@status_message, current_user)
```

Comment:
I think it either should be flat (no associations included) or include all nested associations, currently it's a mixture iirc.